gcpkms/keysource.go

```diff
@@ -151,3 +153,19 @@ func (key MasterKey) ToMap() map[string]interface{} {
 	out["created_at"] = key.CreationDate.UTC().Format(time.RFC3339)
 	return out
 }
+
+// getDefaultApplicationCredentials allows for passing GCP Service Account
+// Credentials as either a path to a file, or directly as an environment variable
+// in JSON format.
+func getDefaultApplicationCredentials() (token []byte, err error) {
+	var defaultCredentials = os.Getenv("GOOGLE_CREDENTIALS")
```

Comment:
I believe this should be `GOOGLE_APPLICATION_CREDENTIALS`?

Reply:
One of the goal of this pull request is to be able to read credentials directly out of an environment variable; `GOOGLE_APPLICATION_CREDENTIALS` contains the _path_ to a file containing the credentials, not the credentials itself.

This is not supported natively by the Google SDK (I can find the related issues in their repository, if needed), but other consumers may provide this, for example the [Terraform Google provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#using-terraform-cloud). Being able to read the credentials directly out of an environment variable would open up the possibility to use sops in context where it's not possible to read the credentials otherwise.

Hence, the deliberate use of this `GOOGLE_CREDENTIALS` environment variable, like the Google provider for Terraform does.

Reply:
Yes agreed, thank you @multani. 

Apologies, I misread your original [comment](https://github.com/mozilla/sops/pull/953#issuecomment-996788870). I've updated these changes to support using either `GOOGLE_CREDENTIALS` or `GOOGLE_APPLICATION_CREDENTIALS` such that if a key or path isn't provided in `GOOGLE_CREDENTIALS` it will fallback to the expected behavior from `cloudkms.NewService()` -- which uses the default application credentials automatically.